### ISSUE TITLE
How to handle `grunt` properties/methods.

### ISSUE THREAD

**Issue by cowboy** [gruntjs/grunt-legacy-file#4]
Looking at the uses of `grunt.*` methods I see:

* `grunt.util`
 * We should probably just require lodash directly for `_` and not grab it from `util`, we just need to make sure that if we use a newer version of lodash, that our unit tests are comprehensive in case something changed with any of the methods we use.
 * I believe both `util.kindOf` and `util.toArray` are in lodash as well (probably with different names). If so, let's just use them right from lodash.
 * This might leave us only needing `util.error`, which we can get right from `grunt-legacy-util`.
* `grunt.option`
 * This needs to be handled with a `prototype.option` method that uses the passed grunt or uses passed-in options, just like [grunt-legacy-log](https://github.com/gruntjs/grunt-legacy-log/blob/master/index.js#L119-L127).
* `grunt.verbose`
 * Seeing as how this module only logs using `grunt.verbose` we can either grab this from the passed-in `grunt` or via a `log` option. We can create a method for logging on the prototype that does something like:
```js
File.prototype.log = function() {
  var log = this.options.grunt ? this.options.grunt.log.verbose : this.options.log;
  return log.apply(this, arguments);
};
```
* `grunt.log`
 * Like `grunt.verbose`, seeing as how this module only uses `grunt.log.warn` we can make a method for warning that does something like:
```js
File.prototype.warn = function() {
  var warn = this.options.grunt ? this.options.grunt.log.warn : this.options.warn;
  return warn.apply(this, arguments);
};
```
* `grunt.fail`
 * Not sure how to handle this one yet. I need to look into what, exactly, `grunt.fail.warn` is doing.

**Comment by jonschlinkert** [comment]
> we just need to make sure that if we use a newer version of lodash,

yes, the first thing that comes to mind (and there are others) is how arguments are passed to `_.template`. still reading over the rest...

**Comment by jonschlinkert** [comment]
couple of questions.

since lodash is only used for two things, `_.toArray` and `kindOf`...

> util.toArray 

Would `[].slice.call(arguments)` be okay to use there (or even an inline `for` loop if we care about any tiny v8 optimizations we might get), or is there an additional advantage of using `_.toArray`? I started skimming the lodash source for `_.toArray` then realized there might just be use cases here that I'm oblivious to.

> util.kindOf

If the first question is doable, would you be okay with using https://www.npmjs.com/package/kind-of instead of lodash for `kindOf`? this way we can make the deps a little lighter

for now I'll use lodash

**Comment by jonschlinkert** [comment]
I had a question up here about lodash that was based on your suggestion about `toArray` and `kindOf`... but at first I thought you said those were the only two methods used from lodash. ignore this lol

**Comment by jonschlinkert** [comment]
another todo:

- [  ] extend options: https://github.com/gruntjs/grunt-legacy-file/pull/3#discussion_r30844622

**Comment by cowboy** [comment]
* For now, let's use `util.error` from grunt-legacy-util, so we can maintain that code in just once place.
* Regarding `util.kindOf` we can just use `_.isObject` ([see docs](https://lodash.com/docs#isObject)) which performs the test we need.
* Regarding `grunt.verbose`, I think it would be wise to not depend on any other "legacy" libs (other than util) directly, but instead pass them into the constructor. This should simplify testing and maintenance.
* Same with `grunt.log`.

**Comment by jonschlinkert** [comment]
@cowboy fwiw:

```js
_.isObject(function () {});
//=> true
_.isObject(/foo/)
//=> true
```
this is probably why you used `util.kindOf` before.
